h2o-automl/src/main/java/water/automl/api/LeaderboardsHandler.java

```diff
@@ -42,30 +43,29 @@ public boolean filter(KeySnapshot.KeyInfo k) {
   }
 
   @SuppressWarnings("unused") // called through reflection by RequestServer
-  public LeaderboardV99 fetch(int version, LeaderboardsV99 s) {
-    if (null == s.project_name)
+  public LeaderboardV99 fetch(int version, LeaderboardsV99 input) {
+    if (null == input.project_name)
       throw new H2OKeyNotFoundArgumentException("Client must specify a project_name.");
-
-    return new LeaderboardV99().fillFromImpl(getFromDKV("project_name", Leaderboard.idForProject(s.project_name)));
+    Leaderboard leaderboard = getFromDKV(Key.make(Leaderboard.idForProject(input.project_name)), "project_name");
+    LeaderboardV99 lb = new LeaderboardV99().fillFromImpl(leaderboard);
+    if (input.extensions != null) {
+      lb.table = new TwoDimTableV3().fillFromImpl(leaderboard.toTwoDimTable(input.extensions));
+    }
+    return lb;
   }
 
-  // TODO: almost identical to ModelsHandler; refactor
-  public static Leaderboard getFromDKV(String param_name, String key_str) {
-    return getFromDKV(param_name, Key.make(key_str));
+  private static Leaderboard getFromDKV(Key key) {
+      return getFromDKV(key, "(none)");
```

Comment:
@seb-h2o looks like Java 8 Optional

Reply:
I don't see a big benefit in using `Optional` here. It always needs a string for the exception...

Reply:
@seb-h2o i was not actually suggesting to change anything, just was surprised by this implementation of the Option concept :)